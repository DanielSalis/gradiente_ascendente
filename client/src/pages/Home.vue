<template>
  <v-container class="centered-container">
    <h3>Bem vindo ao SnapLearn!</h3>

    <p class="mt-4">
      Já tem um token da OpenAI? Informe abaixo ou realize um cadastro para acumular pontos
    </p>

    <div class="home__container mt-8">
      <div class="home__container-token">
        <v-text-field
          solo
          hide-details
          label="Token"
          placeholder="Token"
        />
        <v-btn
          color="primary"
          large
        >
          <v-icon>mdi-arrow-right</v-icon>
        </v-btn>
      </div>
      OU
      <v-btn
        color="primary"
        to="/register"
      >
        Criar conta
      </v-btn>

      <v-btn to="/login">
        Já tenho uma conta
      </v-btn>
    </div>
  </v-container>
</template>

<script>
  import {api} from '../utils/api';
  import { mapState } from 'vuex'

  export default {
    name: 'HomePage',
    data(){
      return {
        posts: [],
        errors: []
      }
    },

    computed: {
      ...mapState('test', ['test']),
      users() {
        return this.$store.getters.getTest
      },
    },
    async created(){
      await api.get(`entries`)
        .then(response => {
          console.log(response);
        })
        .catch(e => {
          console.log(e);
        })
    },
  }
</script>

<style lang="scss" scoped>
.home__container {
  display: flex;
  flex-direction: column;
  row-gap: 16px;
}

.home__container-token {
  align-items: center;
  column-gap: 8px;
  display: flex;
}
</style>
