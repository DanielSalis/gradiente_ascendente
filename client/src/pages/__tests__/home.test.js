import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'
import Home from '../../pages/Home.vue'

const localVue = createLocalVue()
// localVue.use(Vuex)

// const store = new Vuex.Store({
//   modules: {
//     city: {
//       namespaced: true,
//       state: {
//         cities: []
//       },
//       actions: {
//         fetchCities: () => []
//       }
//     },
//     cid: {
//       namespaced: true,
//       state: {
//         cids: []
//       },
//       actions: {
//         fetchCids: () => []
//       }
//     }
//   }
// })

describe('Pages / Home', () => {
  let vuetify
  let router
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
    router = new VueRouter()

    wrapper = mount(Home, {
      localVue,
      vuetify,
      router,
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
  })

  describe('when all data is correct', () => {
    it('should render the page', () => {
      expect(wrapper.element).toMatchSnapshot();
    })
    it('renders the correct div ', ()=>{
      expect(wrapper.find('.home__container').exists()).toBe(true)
    })
  })
})
