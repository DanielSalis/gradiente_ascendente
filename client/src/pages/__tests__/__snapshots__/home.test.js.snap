// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages / Home when all data is correct should render the page 1`] = `
<div
  class="container"
>
  <h3>
    Bem vindo a ferramenta!
  </h3>
   
  <p
    class="mt-4"
  >
    
    Já tem um token da OpenAI? Informe abaixo ou realize um cadastro para conseguir acumular pontos
  
  </p>
   
  <div
    class="home__container mt-8"
  >
    <div
      class="home__container-token"
    >
      <div
        class="v-input v-input--hide-details theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder"
      >
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
          >
            <div
              class="v-text-field__slot"
            >
              <label
                class="v-label theme--light"
                for="input-2"
                style="left: 0px; position: absolute;"
              >
                Token
              </label>
              <input
                id="input-2"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
       
      <button
        class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--large primary"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-arrow-right theme--light"
          />
        </span>
      </button>
    </div>
    
    OU
    
    <a
      class="v-btn v-btn--is-elevated v-btn--has-bg v-btn--router theme--light v-size--default primary"
    >
      <span
        class="v-btn__content"
      >
        
      Criar conta
    
      </span>
    </a>
     
    <button
      class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        Já tenho uma conta
      </span>
    </button>
  </div>
</div>
`;
